src/dune_rules/dune_package.ml

```diff
@@ -133,10 +135,19 @@ module Lib = struct
        and+ plugins = mode_paths "plugins"
        and+ foreign_objects = paths "foreign_objects"
        and+ foreign_archives =
-         if lang.version >= (2, 0) then paths "foreign_archives"
+         if lang.version >= (3, 5) then
```

Comment:
I am not sure I understand your comment. I have to change the format of the Dune package files to handle mode-specific archives. It is easier to use this new format even when there is no mode-specific stubs so the only criteria here it to check which version of Dune wrote the dune-package field to decode it properly.

Reply:
I'm talking about the case where the lang is 2.0 and the per-mode syntax is used. the error message won't mention that it's necessary to upgrade dune lang to 3.5 (actually, this path should now be updated to use the extension syntax instead of dune lang). Can you add a test case for this? It'll be easier to see what I mean with a concrete error message.

Reply:
ok I had missed that this was in `dune-package`. I'm not sure how versioning is supposed to work for these files. @rgrinberg can you have a look at this bit of the PR?

Reply:
come to think of it, it seems fine.

Reply:
There is no versioning for these files as they're re-generated by dune anyway.

Reply:
that's what I figured. I wonder why there's a need to distinguish on lang.version then. it looks like they are implicitly versioned by the version of the dune binary itself. the only case this matters is when developing dune probably?